core/server/models/base/index.js

```diff
@@ -712,7 +712,10 @@ ghostBookshelf.Model = ghostBookshelf.Model.extend({
             });
         };
 
-        slug = utils.safeString(base, options);
+        // the slug may never be longer than the allowed limit of 191 chars, but should also
+        // take the counter into count. We reduce a too long slug to 185 so we're always on the
+        // safe side, also in terms of checking for existing slugs already.
+        slug = utils.safeString(base, options).slice(0, 185);
```

Comment:
The only case i can think of which is very edgy, but still want to share.
If you import a slug which is longer than 185, we don't want to cut off the slug.
Instead we want to show the validation error to the user, so he can fix his issue by himself.

Other than that, i have no simpler or better idea to this. It's edgy anyway.

Reply:
I‘ll check tomorrow, if this rn is actually ever touched when importing. Currently I‘d say no, but I‘ll verify this tomorrow and maybe create an error if so.

Reply:
Thanks 👍  Other than that we can merge this, thanks.

Reply:
Unfortunately, they are cut off on an import 😔

I played around with it for a couple of hours and couldn't find an appropriate solution. The only easy-to-implement way I found, was to show a warning, rather than an error, [here](https://github.com/TryGhost/Ghost/blob/master/core/server/data/importer/importers/data/base.js#L60)

The only place and way I found where import errors are produces is in the schema validation, which is called [here](https://github.com/TryGhost/Ghost/blob/master/core/server/models/base/index.js#L124). But because, we didn't set a shorter soft limit for the slug values, the validation would only fail, when the hard limit of 191 is reached.

So one easy option for this super edge case would be to produce a warning, when the slug is longer than 185 characters, cut it off anyway, and produce the normal validation error, when the slug is > 191 characters.

Reply:
I changed the code, so that the slug will not be cut on an import.